games-rpg/draci-historie/draci-historie-2012-r1.ebuild

```diff
@@ -1,79 +1,60 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-inherit eutils
+EAPI=8
+
+inherit desktop wrapper
 
 DESCRIPTION="Bert the little dragon searches for his father"
-HOMEPAGE="http://www.ucw.cz/draci-historie/index-en.html"
-BASE_URL="http://www.ucw.cz/draci-historie/binary/dh"
+HOMEPAGE="https://www.ucw.cz/draci-historie/index-en.html"
+BASE_URL="https://www.ucw.cz/draci-historie/binary/dh"
 SRC_URI="
 	l10n_cs? ( ${BASE_URL}-cz-${PV}.zip )
 	l10n_de? ( ${BASE_URL}-de-${PV}.zip )
 	l10n_en? ( ${BASE_URL}-en-${PV}.zip )
 	l10n_pl? ( ${BASE_URL}-pl-${PV}.zip )
-	!l10n_cs? ( !l10n_de? ( !l10n_en? ( !l10n_pl? ( ${BASE_URL}-en-${PV}.zip ) ) ) )
 "
+S="${WORKDIR}"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 IUSE="l10n_cs l10n_de l10n_en l10n_pl"
+REQUIRED_USE="|| ( l10n_cs l10n_de l10n_en l10n_pl )"
 
 RDEPEND=">=games-engines/scummvm-1.1"
-DEPEND="app-arch/unzip"
-
-S="${WORKDIR}"
+BDEPEND="app-arch/unzip"
 
 src_unpack() {
-	if use l10n_en || ( ! use l10n_cs && ! use l10n_de && ! use l10n_en && ! use l10n_pl ) ; then
-		mkdir en || die
-		unpack dh-en-${PV}.zip
-		mv *.{dfw,fon,mid,sam} en/ || die
-	fi
-	if use l10n_cs ; then
-		mkdir cs || die
-		unpack dh-cz-${PV}.zip
-		mv *.{dfw,fon,mid,sam,zzz} cs/ || die
-	fi
-	if use l10n_de ; then
-		mkdir de || die
-		unpack dh-de-${PV}.zip
-		mv *.{dfw,fon,mid,sam} de/ || die
-	fi
-	if use l10n_pl ; then
-		mkdir pl || die
-		unpack dh-pl-${PV}.zip
-		mv *.{dfw,fon,mid,sam,zzz} pl/ || die
-	fi
+	local loc
+	for loc in cs de en pl; do
+		if use l10n_${loc}; then
+			mkdir ${loc} || die
+			unpack dh-${loc}-${PV}.zip
+			mv *.{dfw,fon,mid,sam} ${loc}/ || die
+		fi
+	done
 }
 
 src_prepare() {
 	default
-	rm -f *.{bat,exe,ins} readme.* || die
+	rm -f *.{bat,exe,ins,ico} readme.* || die
 }
 
 src_install() {
-	newicon bert.ico draci-historie.ico
 	insinto /usr/share/${PN}
+	local lingua
 	for lingua in $(find * -type d); do
 		doins -r ${lingua}
 	done
```

Comment:
Since we have our own loop, just remove this block and add a `doins -r ${loc}` in the new one.